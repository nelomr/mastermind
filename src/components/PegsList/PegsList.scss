.row-board--pegs {
  display: flex;
  flex-wrap: wrap;
  max-width: 40px;
  margin: 0 -5px;
}

.peg-item {
  width: 10px;
  height: 10px;
  margin: 0 5px 5px;
  border-radius: 50%;
}