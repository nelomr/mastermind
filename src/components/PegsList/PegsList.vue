<template>
  <div class="row-board--pegs">
    <PegItem 
      v-for="(color, index) in pegsColors" 
      :key="'pegs'+index"
      :color="color"
    />
  </div>
</template>
<script>
import { mapState } from 'vuex';
import PegItem from '@/components/PegItem/PegItem'
export default {
  name: 'PegsList',
  components: {
    PegItem
  },
  props: {
    numSlots: {
      type: Number,
      default: 0
    },
    indexRow: {
      type: Number,
      default: 0
    }
  },
  computed: {
    ...mapState({
      game: state => state.game,
      row: state => state.row,
      pegsArray: state => state.pegsColors
    }),
    pegsColors() {
      //console.log(this.indexRow, this.row, this.pegsArray);
      if(this.indexRow == (this.row - 1)) {
        return this.pegsArray.length > 0 ? this.pegsArray : this.defaultColorsArray;
      }
      return this.defaultColorsArray;
    },
    defaultColorsArray() {
      let array = [];
      for (let index = 0; index < this.numSlots; index+=1) {
        array.push('gray');
      }
      return array;
    }
  }
}
</script>
<style lang="scss">
  @import "./PegsList.scss";
</style>