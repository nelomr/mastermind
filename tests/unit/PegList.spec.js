import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PegsList from '@/components/PegsList/PegsList.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('PegsList.vue', () => {
  let pegsColor = ['lightgray','lightgray','lightgray','lightgray'];
  let rowsPegs;
  let pegsColorDefault = 'lightgray';
  let indexRow = 0;

  let props;
  let store;
  let state;

  beforeEach(() => {
    props = {
      indexRow
    };
    state = {
      pegsColor,
      rowsPegs: pegsColor,
      pegsColorDefault
    };
    store = new Vuex.Store({
      state
    });
  });
  const build = () => {
    const wrapper = shallowMount(PegsList, {
      props,
      mocks: {$store: store}
    });

    return {
        wrapper
    };
  };

  it('renders the component list', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
})
