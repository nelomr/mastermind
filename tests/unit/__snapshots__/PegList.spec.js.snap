// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PegsList.vue renders the component list 1`] = `<div class="row-board--pegs"></div>`;
